tools/duplicate_function_eliminator.py

```diff
@@ -0,0 +1,428 @@
+
```

Comment:
At some level, this is also fallout from the fact that we separated out the DFE code from js_optimizer.py since we were previously passing in various args in js_optimizer.py to minify_shell to tell it to perform DFE rather than minification.